dev/how-to/maintain/tidb-in-kubernetes/log-collecting.md

```diff
@@ -0,0 +1,97 @@
+
+[ElasticSearch](https://kubernetes.io/docs/tasks/debug-application-cluster/logging-elasticsearch-kibana/) and [Stackdriver](https://kubernetes.io/docs/tasks/debug-application-cluster/logging-stackdriver/) are provided as two log collection approaches for reference in the Kubernetes official documentation.
+
+Common open source tools that can be used to collect Kubernetes logs are:
+
+- [Fluentd](https://www.fluentd.org/)
+- [Fluent-bit](https://fluentbit.io/)
+- [Filebeat](https://www.elastic.co/products/beats/filebeat)
+- [Logstash](https://www.elastic.co/products/logstash)
+
+Collected Logs can usually be aggregated and stored on a specific server or stored in a dedicated storage and analysis system such as ElasticSearch.
+
+Some cloud service providers or specialized performance monitoring service providers also have their own free or chargeable log collection options that you can choose.
+
+If you do not aggregate logs via a separate log collection tool, you can also use the `kubectl` tool to directly view the runtime log of a container, but you cannot view the log of a destroyed container with this approach:
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl logs -n ${namespace} ${tidbPodName}
+```
+
+> **Note:**
+>
+> To view the log before the container restarts, you can add the `-p` parameter when executing the above command.
+
+If you need to collect logs from multiple Pods, you can use [`stern`](https://github.com/wercker/stern):
+
+{{< copyable "shell-regular" >}}
+
+```shell
+stern -n ${namespace} tidb -c slowlog
+```
+
+## Collect TiDB slow query logs
+
+For versions prior to 3.0, by default, TiDB prints slow query logs to standard output, mixed with application logs.
+
+- For the TiDB version <= 2.1.7, you can filter the slow query logs with the keyword `SLOW_QUERY`, for example:
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl logs -n ${namespace} ${tidbPodName} | grep SLOW_QUERY
+    ```
+
+- For the TiDB version >= 2.1.8, it is not so easy to separate the slow query log due to the change of the slow query log format. For this reason, it is recommended to configure `separateSlowLog: true` as described below to view the slow query log separately.
+
+In some cases, you may want to use some tools or automated systems to analyze and process the log content. The application log of each TiDB component uses [unified log format](https://github.com/tikv/rfcs/blob/master/text/2018-12-19-unified-log-format.md), which facilitates parsing with other programs. But because slow query logs use a multi-line format that is compatible with MySQL, it can be difficult to parse it with slow query logs mixed with application logs.
```

Comment:
```suggestion
In some cases, you may want to use some tools or automated systems to analyze and process the log content. The application log of each TiDB component uses [unified log format](https://github.com/tikv/rfcs/blob/master/text/2018-12-19-unified-log-format.md), which facilitates parsing with other programs. However, because slow query logs use a multi-line format that is compatible with MySQL, it might be difficult to parse slow query logs when they are mixed with application logs.
```